File: csharp/leap/Leap.cs
﻿using System;

namespace leap
{
	public static class Year
	{
		public static bool IsLeap(int year)
		{
			return (year % 400 == 0 ) || ( year % 100 != 0 && year % 4 == 0);
		}
	}
}

File: ruby/leap/leap.rb
class Year
  VERSION = 1

  def self.leap?(year)
    Year.new(year).leap?
  end

  attr_reader :year

  def initialize(year)
    @year = year
  end

  def leap?
    divisible?(4) && !divisible?(100) || divisible?(400)
  end

  private

  def divisible?(n)
    year % n == 0
  end
end